api/app/Http/Controllers/ApiController.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: redjik
+ * Date: 16.07.15
+ * Time: 0:37
+ */
+
+namespace App\Http\Controllers;
+
+use App\Helpers\RouteHelper;
+use App\Repositories\BaseRepository;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\Request;
+use Spira\Repository\Model\BaseModel;
+use Spira\Responder\Responder\ApiResponder;
+use Symfony\Component\HttpFoundation\Response;
+
+abstract class ApiController
+{
+    /**
+     * Model Repository.
+     *
+     * @var BaseRepository
+     */
+    protected $repository;
+
+    /**
+     * @var ApiResponder
+     */
+    protected $responder;
+
+    /**
+     * Get all entities.
+     *
+     * @return Response
+     */
+    public function getAll()
+    {
+        return $this->responder->collection($this->repository->all());
+    }
+
+    /**
+     * Get one entity.
+     *
+     * @param  string $id
+     * @return Response
+     */
+    public function getOne($id)
+    {
+        return $this->responder->item($this->repository->find($id));
+    }
+
+    /**
+     * Post a new entity.
+     *
+     * @param  Request $request
+     * @return Response
+     */
+    public function postOne(Request $request)
+    {
+        $model = $this->repository->getModel();
+        $model->fill($request->all());
+        $this->repository->save($model);
+        $response = $this->responder->created();
+        if ($route = RouteHelper::getRoute($model)) {
+            $response->setContent(json_encode([$route]));
+        }
+        return $response;
+    }
+
+    /**
+     * Put an entity.
+     *
+     * @param  string   $id
+     * @param  Request  $request
+     * @return Response
+     */
+    public function putOne($id, Request $request)
+    {
+        try {
+            $model = $this->repository->find($id);
+        } catch (ModelNotFoundException $e) {
+            $model = $this->repository->getModel();
+        }
+        $model->fill($request->all());
+        $this->repository->save($model);
+
+        $response = $this->responder->created();
+        if ($route = RouteHelper::getRoute($model)) {
+            $response->setContent(json_encode([$route]));
+        }
+
+        return $response;
+    }
+
+    /**
+     * Put many entities.
+     *
+     * @param  Request  $request
+     * @return Response
+     */
+    public function putMany(Request $request)
+    {
+        $data = $request->data;
+        $idName = $this->repository->getModel()->getKeyName();
+        $ids = [];
+        foreach ($data as $piece) {
+            $ids[] = $piece[$idName];
+        }
+        $models = $this->repository->findMany($ids);
```

Comment:
As this section of code above is reused, I think it would make sense to abstract it out to a function eg `$entityIds = $this->getIdsFromRequestCollection($request);`